app/edge/devices/components/AMTDevicesDatatable/AMTDevicesDatatable.tsx

```diff
@@ -0,0 +1,113 @@
+import { useTable, usePagination } from 'react-table';
+
+import {
+  Table,
+  TableContainer,
+  TableHeaderRow,
+  TableRow,
+} from '@/portainer/components/datatables/components';
+import { Checkbox } from '@/portainer/components/form-components/Checkbox';
+import { Device } from '@/portainer/hostmanagement/open-amt/model';
+import { useEnvironment } from '@/portainer/environments/useEnvironment';
+import { useAMTDevices } from '@/edge/devices/components/AMTDevicesDatatable/useAMTDevices';
+import { RowProvider } from '@/edge/devices/components/AMTDevicesDatatable/columns/RowContext';
+
+import { useColumns } from './columns';
+
+export function AMTDevicesDatatable() {
+  const columns = useColumns();
+
+  const environment = useEnvironment();
```

Comment:
this hook was meant to be used as a global environment state. I feel it's misplaced here. Why not get environment (or env id if only this is necessary) as a prop?. 

if any of the columns needs it, just pass it in row context state (RowProvider will recieve it as a prop)

Reply:
done